lib/style/module_directives.ex

```diff
@@ -86,110 +130,68 @@ defmodule Styler.Style.ModuleDirectives do
         {directive, _, _} -> directive
       end)
 
-    # TODO: (optimization)
-    # now that we have use/import/alias/require, we might as well run
-    # them through the sort/expand/dedupe functionality and skip them in the traversal
     shortdocs = directives[:"@shortdoc"] || []
-    moduledocs = directives[:"@moduledoc"] || if needs_moduledoc?(name), do: [@moduledoc_false], else: []
-    # TODO sort behaviours?
-    behaviours = directives[:"@behaviour"] || []
-    behaviours = List.update_at(behaviours, -1, &set_newlines(&1, 2))
+    moduledocs = directives[:"@moduledoc"] || if add_moduledoc?, do: [@moduledoc_false], else: []
+    behaviours = expand_and_sort(directives[:"@behaviour"] || [])
+
+    uses =
+      case directives[:use] do
+        nil ->
+          []
+
+        uses ->
+          # this is `expand_and_sort` except we don't want to sort `use`, so it's just `expand` :)
+          uses
+          |> Enum.flat_map(&expand_directive/1)
+          |> Enum.map(&set_newlines(&1, 1))
+          |> List.update_at(-1, &set_newlines(&1, 2))
+      end
 
-    uses = directives[:use] || []
-    imports = directives[:import] || []
-    aliases = directives[:alias] || []
-    requires = directives[:require] || []
+    imports = expand_and_sort(directives[:import] || [])
+    aliases = expand_and_sort(directives[:alias] || [])
+    requires = expand_and_sort(directives[:require] || [])
 
-    children =
+    directives =
       Enum.concat([
         shortdocs,
         moduledocs,
         behaviours,
         uses,
         imports,
         aliases,
-        requires,
-        other
+        requires
       ])
 
-    zipper = {{:defmodule, def_meta, [name, [{mod_do, {:__block__, children_meta, children}}]]}, zipper_meta}
-    {:cont, zipper, ctx}
-  end
-
-  # a module whose only child is a moduledoc. pass it on through
-  def run({{:defmodule, _, [_, [{_, {:@, _, [{:moduledoc, _, _}]}}]]}, _} = zipper, ctx), do: {:cont, zipper, ctx}
+    parent = Zipper.update(parent, &Zipper.replace_children(&1, directives))
 
-  def run({{:defmodule, def_meta, [name, [{mod_do, mod_children}]]}, zipper_meta} = zipper, ctx) do
-    # a module with a single child. lets add moduledoc false
-    # ... unless it's a `defmodule Foo, do: ...`, that is
-    if needs_moduledoc?(name, mod_do) do
-      mod_children = {:__block__, [], [@moduledoc_false, mod_children]}
-      {:cont, {{:defmodule, def_meta, [name, [{mod_do, mod_children}]]}, zipper_meta}, ctx}
+    if Enum.empty?(nondirectives) do
+      parent
     else
-      {:cont, zipper, ctx}
+      {last_directive, meta} = parent |> Zipper.down() |> Zipper.rightmost()
+      {last_directive, %{meta | r: nondirectives}}
```

Comment:
imo we're better off reaching into the meta here and just shoving in the new siblings. using `insert_right` is just extra enumerations that are totally unnecessary
```suggestion
     nondirectives
     |> Enum.reverse()
     |> Enum.reduce(parent |> Zipper.down() |> Zipper.rightmost(), &Zipper.insert_right(&2, &1))
```